src/test/java/io/openliberty/tools/intellij/it/TestUtils.java

```diff
@@ -562,4 +571,41 @@ public static void copyDirectory(String sourceDirectoryLocation, String destinat
                 });
     }
 
+    /**
+     * Test to see if there has been a fatal error and JUnit should be stopped.
+     * This searches the output of this JUnit run for SocketTimeoutException which
+     * has been identified as a fatal error and occurs during the Mac tests.
+     * It is set up as a static reader so that we do not reread the whole file
+     * after each test.
+     */
+    final static String outputFile = System.getenv("JUNIT_OUTPUT_TXT");
+    final static BufferedReader reader;
+
+    static {
+        try {
+            reader = new BufferedReader(new InputStreamReader(new FileInputStream(outputFile)));
+        } catch (FileNotFoundException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public static void detectFatalError() {
+        try {
+            // Flush our output and then wait briefly for the process running 'tee' to flush output to the
+            // file that the buffered reader is reading.
+            System.out.flush();
+            Thread.sleep(10);
+            // Continue reading the existing reader
```

Comment:
Does this actually work? Once you've reached the end of a stream, in general you can't read any more data from it. This is different from reading from a stream which blocks on read while it waits for the producer to generate more data.

Reply:
I'm not sure how to test whether it works or not. Should I just take it out?

Reply:
What does taking it out mean? Replacing the "optimization" with a locally created Reader and avoiding this and the other issues I mentioned that come with managing it as a static field?